src/vstart.sh

```diff
@@ -736,6 +742,57 @@ $extra_conf
 EOF
 }
 
+write_logrotate_conf() {
+    mon_logs=$(pwd)"/out/mon.*.log"
+    osd_logs=$(pwd)"/out/osd.*.log"
+    mgr_logs=$(pwd)"/out/mgr.*.log"
+    mds_logs=$(pwd)"/out/mds.*.log"
+
+    mon_pids=
+    osd_pids=$(ps -e | grep ceph-osd | awk '{print $1}' | tr '\n' ' ')
+    mds_pids=$(ps -e | grep ceph-mds | awk '{print $1}' | tr '\n' ' ')
+    mgr_pids=$(ps -e | grep ceph-mgr | awk '{print $1}' | tr '\n' ' ')
+
+    cat << EOF
+$mon_logs
+$osd_logs
+$mgr_logs
+$mds_logs
+{
+    copytruncate
+    missingok
+    size 2G
+    rotate 20
+    prerotate
+        kill -s stop \$(ps -e | grep ceph-mon | awk '{print \$1}' | tr '\n' ' ')
+        kill -s stop \$(ps -e | grep ceph-osd | awk '{print \$1}' | tr '\n' ' ')
+        kill -s stop \$(ps -e | grep ceph-mgr | awk '{print \$1}' | tr '\n' ' ')
+        kill -s stop \$(ps -e | grep ceph-mds | awk '{print \$1}' | tr '\n' ' ')
+    endscript
+    postrotate
+        kill -s cont \$(ps -e | grep ceph-mon | awk '{print \$1}' | tr '\n' ' ')
+        kill -s cont \$(ps -e | grep ceph-osd | awk '{print \$1}' | tr '\n' ' ')
+        kill -s cont \$(ps -e | grep ceph-mgr | awk '{print \$1}' | tr '\n' ' ')
+        kill -s cont \$(ps -e | grep ceph-mds | awk '{print \$1}' | tr '\n' ' ')
+    endscript
```

Comment:
Simplify to:

````
/home/pdonnell/ceph/build/out/*.log
{
    copytruncate
    missingok
    sharedscripts
    size 2G
    rotate 20
    postrotate
        killall -u pdonnell -q -1 ceph-mon ceph-mgr ceph-mds ceph-osd ceph-fuse radosgw rbd-mirror
    endscript
}
````

(Fill in the username when writing the conf file.)

Keep in mind we will do this often on shared dev machines so we don't want to see the error messages for sending signals to other user's processes.

Reply:
> Keep in mind we will do this often on shared dev machines so we don't want to see the error messages for sending signals to other user's processes.

Right, agreed.

But you did not specify any signal name which implies you are terminating Ceph process (right?). The intention for my pre/postrotate script was to freeze Ceph process while rotating so that no log entries are missed.

Reply:
`-1` is `-SIGHUP`

Reply:
Simplied, thanks!